### DISCUSSION TITLE
How to deploy to GitHub Pages with GitHub Actions?

### DISCUSSION THREAD

**Discussion by ngoohebi** [google/docsy#1929] [General]
Seems not working with this?
https://github.com/peaceiris/actions-hugo

Any other method?

**Comment by fekete-robert** [comment]
What's the actual issue/error message? 
AFAIK the action you've linked doesn't do deployment, just builds your site, you have to do deployment in a separate step. For an example, see: https://github.com/bank-vaults/bank-vaults.dev/blob/main/.github/workflows/publish.yaml

**Comment by deining** [comment]
> Seems not working with this? https://github.com/peaceiris/actions-hugo
> 
> Any other method?

You may have a look at the [preview](https://deploy-preview-1435--docsydocs.netlify.app/docs/deployment/#deployment-on-github-pages) of my open PR #1435, explaining how to deploy to GitHub Pages with GitHub Actions.

In the docsy example repo, there exists [PR 234](https://github.com/google/docsy-example/pull/234) that includes a [workflow file](https://github.com/google/docsy-example/blob/8f081fb5bf9ab1479232a83959b70bf8e895b765/.github/example-workflows/gh-pages.yml) that exactly does what you are trying to achieve.

**Reply by ngoohebi** [reply]
@deining Hello, thanks for the info.

I tried the `deploy-github-pages.yml` file from the preview docs but failed.

Error:
```
Found in cache @ /opt/hostedtoolcache/node/20.12.1/x64

Environment details

  node: v20.12.1
  npm: 10.5.0
  yarn: 1.22.22
/opt/hostedtoolcache/node/20.12.1/x64/bin/npm config get cache
/home/runner/.npm
Error: Some specified paths were not resolved, unable to cache dependencies.
```

Did I miss something?

**Reply by ngoohebi** [reply]
@deining The [workflow file](https://github.com/google/docsy-example/blob/8f081fb5bf9ab1479232a83959b70bf8e895b765/.github/example-workflows/gh-pages.yml) did its job with no problem.
